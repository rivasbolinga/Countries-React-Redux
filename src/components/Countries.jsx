import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Countries = ({
  num, flag, name, capital, population, region,
}) => (
  <Wrapper className="country-card">
    <Link to={`/country/${num}`} className="link">
      <img src={flag.png} alt={flag.alt} className="flag" />
      <h2>{name}</h2>
      <p>
        <span>Capital:</span>
        {' '}
        {capital}
      </p>
      <p>
        <span>Population:</span>
        {' '}
        {population}
      </p>
      <p>
        <span>Region:</span>
        {' '}
        {region}
      </p>
    </Link>
  </Wrapper>
);

Countries.propTypes = {
  num: PropTypes.number.isRequired,
  flag: PropTypes.shape({
    png: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
  }).isRequired,
  name: PropTypes.string.isRequired,
  capital: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
  region: PropTypes.string.isRequired,
};

const Wrapper = styled.section`
  border: 1px solid;
  .flag {
    width: 200px;
    height: auto;
  }
}

`;

export default Countries;
