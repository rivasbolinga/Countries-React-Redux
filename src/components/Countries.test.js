import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Countries from './Countries';

describe('Countries component', () => {
  test('renders correctly', () => {
    const props = {
      num: 1,
      flag: { png: 'flag.png', alt: 'Flag' },
      name: 'Country Name',
      capital: 'Capital City',
      population: '1000000',
      region: 'Region',
    };
    const tree = renderer
      .create(
        <MemoryRouter>
          <Countries
            num={props.num}
            flag={props.flag}
            name={props.name}
            capital={props.capital}
            population={props.population}
            region={props.region}
          />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
