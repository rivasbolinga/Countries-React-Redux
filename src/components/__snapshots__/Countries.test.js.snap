// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries component renders correctly 1`] = `
<div
  className="sc-beqWaB bBCdOB country-card"
>
  <a
    className="link"
    href="/country/1"
    onClick={[Function]}
  >
    <img
      alt="Flag"
      className="flag"
      src="flag.png"
    />
    <div
      className="text"
    >
      <h2>
        Country Name
      </h2>
      <p>
        <span>
          Capital:
        </span>
         
        Capital City
      </p>
      <p>
        <span>
          Continent:
        </span>
         
        Region
      </p>
    </div>
  </a>
</div>
`;
